src/Auth/SeedDatabase.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Codidact.Infrastructure.Identity;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Identity;
+using Microsoft.Extensions.DependencyInjection;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace Codidact.Auth
+{
+    public static class SeedDatabase
+    {
+        public async static Task InitializeIdentityUsers(this IApplicationBuilder app)
+        {
+            var toCreateApplicationUsers = new List<ApplicationUser>
+            {
+                new ApplicationUser
+                {
+                    UserName= "admin",
+                    Email = "admin@codidact.com",
+                }
+            };
+            using var serviceScope = app.ApplicationServices.GetRequiredService<IServiceScopeFactory>().CreateScope();
+            using var userManager = serviceScope.ServiceProvider.GetService<UserManager<ApplicationUser>>();
+            foreach (var user in toCreateApplicationUsers)
+            {
+                var exists = await userManager.FindByNameAsync(user.UserName);
+                if (exists == null)
+                {
+
+                    var randomPassword = GenerateRandomPassword();
+                    var result = await userManager.CreateAsync(user, randomPassword);
+                    if (result.Succeeded)
+                    {
+                        Console.WriteLine($"User created :{user.UserName}. Password: ${randomPassword}. Please change your password as soon as possible.");
+                    }
+                    else
+                    {
+                        Console.WriteLine($"Error Creating user:");
+                        foreach (var error in result.Errors)
+                        {
+                            Console.WriteLine($"({error.Code}) {error.Description}");
+                        }
+
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Generates a Random Password
+        /// respecting the given strength requirements.
+        /// </summary>
+        /// <param name="opts">A valid PasswordOptions object
+        /// containing the password strength requirements.</param>
+        /// <returns>A random password</returns>
+        private static string GenerateRandomPassword()
```

Comment:
**This function is a security concern.**

 * First and foremost - if C# provides a native function to generate a random password, use that instead. Never roll your own security when tested implementations are available. [`System.Web.Security.Membership.GeneratePassword`](https://docs.microsoft.com/en-us/dotnet/api/system.web.security.membership.generatepassword?redirectedfrom=MSDN&view=netframework-4.8#System_Web_Security_Membership_GeneratePassword_System_Int32_System_Int32_) may fit that description. Failing that, the following will make this more secure:

 * Using `Environment.TickCount` to seed a `Random` instance may be acceptable for non-security use-cases, but particularly in password generation it represents a vulnerability because it's predictable.

 * Security-conscious functions should use a secure random number generator; `System.Random` is a PRNG, not a csPRNG. Use `System.Security.Cryptography.RNGCryptoServiceProvider` or equivalent instead:

    ```c#
    // Put provider somewhere accessible to everything that needs random values.
    var provider = new System.Security.Cryptography.RNGCryptoServiceProvider();

    var randomBytes = new byte[4];
    provider.GetBytes(randomBytes);

    var randomInteger = BitConverter.ToUInt32(randomBytes, 0);
    ```

 * The way this method is written inserts required character types in a predictable order, which reduces the effective entropy of the resulting password. Instead of checking whether a character type is required and inserting one if so, try something like this:

    ```c#
    string[] randomChars = new[] { ... }; // as you had it already
    string availableChars = "";
    
    if (opts.RequireUppercase)
    {
        availableChars += randomChars[0];
    }
    
    if (opts.RequireLowercase) 
    {
        availableChars += randomChars[1];
    }
    
    // etc
    // continue with your for loop unchanged
    ```

Reply:
> First and foremost - if C# provides a native function to generate a random password, use that instead. Never roll your own security when tested implementations are available. System.Web.Security.Membership.GeneratePassword may fit that description.

.net core doesn't. Generating passwords is a security flaw because 
>  ...password generation it represents a vulnerability because it's predictable

Maybe just open user creation page if there is no user on launch? And I'll delete the whole class?

Reply:
> Maybe just open user creation page if there is no user on launch?

Not sure, whether that is good for implementation, but from a user point of view running a "setup wizard" when there is no user seems like a good idea. That one could configure:

- The first user account
- The default site settings (i.e. name)
- ...

Reply:
That's the ideal way to do it, yes - I originally suggested just generating a password and printing it because that seemed to be the least amount of work.

Reply:
I enabled the user to input his data via console. Password is invisible https://github.com/codidact/core/pull/29/commits/58e189dd03a7f1d66eb32fc781e696c34d78a6f6